### PR TITLE
Added quick method for JointPlotVisualizer

### PR DESCRIPTION
**Pull request by pdamodaran**
This PR addresses JointPlotVisualizer Quick Method for #960

I have made the following changes:

Added a quick method for JointPlotVisualizer
Added a quick method test to the JointPlotVisualizer tests
Added summary table to the top jointplot.rst
Added an example usage of the quick method to jointplot.rst
Added the quick method to the API Reference

### CODE REVIEW

File: tests/test_features/test_jointplot.py [hunk 1]
```diff
@@ -492,3 +492,18 @@ def test_columns_double_index_continuous_y_hist(self):
...
+    )
+    def test_quick_method(self):
+        """
+        Test the joint_plot quick method
+        """
+        oz = joint_plot(self.continuous.X, self.continuous.y, columns=1, show=False)
+        assert isinstance(oz, JointPlot)
+        assert hasattr(oz, "corr_")
+
+        oz.finalize()
```
Comment (bbengfort) [inline tests/test_features/test_jointplot.py]: This call to finalize shouldn't be necessary; shouldn't the quick method be doing this when `show=False`?

File: tests/test_features/test_jointplot.py [hunk 2]
```diff
@@ -492,3 +492,18 @@ def test_columns_double_index_continuous_y_hist(self):
...
+        IS_WINDOWS_OR_CONDA,
+        reason="font rendering different in OS and/or Python; see #892",
+    )
+    def test_quick_method(self):
+        """
+        Test the joint_plot quick method
+        """
+        oz = joint_plot(self.continuous.X, self.continuous.y, columns=1, show=False)
+        assert isinstance(oz, JointPlot)
+        assert hasattr(oz, "corr_")
```
Comment (bbengfort) [inline tests/test_features/test_jointplot.py]: Great test!

File: yellowbrick/features/jointplot.py [hunk 3]
```diff
@@ -447,5 +447,120 @@ def _index_into(self, idx, data):
...
+        kind="scatter",
+        hist=True,
+        alpha=0.65,
+        joint_kws=None,
+        hist_kws=None,
+        show=True,
+        **kwargs
+):
+    """
+        Joint plots are useful for machine learning on multi-dimensional data, allowing for
```
Comment (bbengfort) [inline yellowbrick/features/jointplot.py]: It looks like there is one extra indent for the docstring, would you mind unindenting it and the final `"""` one level?

File: yellowbrick/features/jointplot.py [hunk 4]
```diff
@@ -447,5 +447,120 @@ def _index_into(self, idx, data):
 ##########################################################################
 
 
-def joint_plot():
-    raise NotImplementedError("quick method still needs to be implemented")
```
Comment (bbengfort) [inline yellowbrick/features/jointplot.py]: Thank you for implementing this!

File: yellowbrick/features/jointplot.py [hunk 5]
```diff
@@ -447,5 +447,120 @@ def _index_into(self, idx, data):
...
+            Specify a transparency where 1 is completely opaque and 0 is completely
+            transparent. This property makes densely clustered points more visible.
+
+        {joint, hist}_kws : dict, default: None
+            Additional keyword arguments for the plot components.
+
+        show : bool, default: True
+        If True, calls ``show()``, which in turn calls ``plt.show()`` however
+        you cannot call ``plt.savefig`` from this signature, nor
+        ``clear_figure``. If False, simply calls ``finalize()``
```
Comment (bbengfort) [inline yellowbrick/features/jointplot.py]: Looks like this needs to be indented in one.

File: tests/test_features/test_jointplot.py (hunk 1 above)
Comment (pdamodaran) [reply tests/test_features/test_jointplot.py]: that's true - I missed that - this has been removed

File: yellowbrick/features/jointplot.py (hunk 3 above)
Comment (pdamodaran) [reply yellowbrick/features/jointplot.py]: the indent issue has been fixed.

File: yellowbrick/features/jointplot.py [hunk 6]
```diff
@@ -447,5 +447,120 @@ def _index_into(self, idx, data):
...
+        Specify a transparency where 1 is completely opaque and 0 is completely
+        transparent. This property makes densely clustered points more visible.
+
+    {joint, hist}_kws : dict, default: None
+        Additional keyword arguments for the plot components.
+
+    show : bool, default: True
+    If True, calls ``show()``, which in turn calls ``plt.show()`` however
+    you cannot call ``plt.savefig`` from this signature, nor
+    ``clear_figure``. If False, simply calls ``finalize()``
```
Comment (bbengfort) [inline yellowbrick/features/jointplot.py]: Still a minor indent issue here - sorry to nit pick; if I get the chance, I'd be happy to fix it.

Comment (pdamodaran) [reply yellowbrick/features/jointplot.py]: oops - I missed that.  this has been fixed.

Comment (bbengfort) [reply yellowbrick/features/jointplot.py]: No worries - thank you!

**Pull request merged**